### PR TITLE
Add BaseRotatingHandler

### PR DESCRIPTION
**Pull request by aminalaee**
https://docs.python.org/3/library/logging.handlers.html#baserotatinghandler

Not sure if we need to test this directly or we can test it with the `RotatingFileHandler ` and others.

### CODE REVIEW

File: src/picologging/handlers.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        """
+        Emit a record.
+
+        Output the record to the file, catering for rollover as described
+        in doRollover().
+        """
+        try:
+            if self.shouldRollover(record):
+                self.doRollover()
+            picologging.FileHandler.emit(self, record)
```
Comment (tonybaloney) [inline src/picologging/handlers.py:49]: Just like in the other PR, this isn't the right way to call parent type methods, I don't understand why the standard library does it like this. 

It should use `super()` and the parent should be initialized correctly. You can be specific in super calls to the parent method to invoke.

**Review by tonybaloney — CHANGES_REQUESTED** [review]

File: src/picologging/handlers.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        handler, if it's callable, passing the source and dest arguments to
+        it. If the attribute isn't callable (the default is None), the source
+        is simply renamed to the destination.
+        :param source: The source filename. This is normally the base
+                       filename, e.g. 'test.log'
+        :param dest:   The destination filename. This is normally
+                       what the source is rotated to, e.g. 'test.log.1'.
+        """
+        if not callable(self.rotator):
+            # Issue 18940: A file may not have been created if delay is True.
```
[the remark this answers is not in the document]
Comment (aminalaee) [reply src/picologging/handlers.py]: Done.

**Comment by tonybaloney** [comment]
Related:  https://github.com/python/cpython/issues/94327

**Review by tonybaloney — APPROVED** [review]

**Comment by tonybaloney** [comment]
Thanks!

**Pull request merged**